controller/api/destination/server.go

```diff
@@ -353,6 +336,65 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	return nil
 }
 
+// sendPodProfile accepts the GetProfile response stream, a pod and its port and
+// returns an error if it cannot stream a profile response back to the client. The
+// profile response sent here is different to a service's since it also includes
+// an endpoint.
+func (s *server) sendPodProfile(stream pb.Destination_GetProfileServer, pod *corev1.Pod, port uint32) error {
```

Comment:
Might consider naming this `sendEndpointProfile` since it applies to all non-service IPs (including pod IPs and non-pod IPs) but will always return an Endpoint in the DestinationProfile.